SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -796,39 +798,58 @@ void USpatialActorChannel::RemoveRepNotifiesWithUnresolvedObjs(TArray<UProperty*
 	});
 }
 
-void USpatialActorChannel::SpatialViewTick()
+void USpatialActorChannel::ServerViewTick()
 {
-	if (Actor != nullptr && !Actor->IsPendingKill() && IsReadyForReplication())
+	if (!IsAuthoritativeServer())
+	{
+		return;
+	}
+
+	UObject* ControllerCDO = AController::StaticClass()->GetDefaultObject();
+	UObject* PlayerControllerCDO = APlayerController::StaticClass()->GetDefaultObject();
+
+	FString OldOwnerWorkerAttribute = NetOwnerWorkerAttribute;
+	NetOwnerWorkerAttribute = improbable::GetOwnerWorkerAttribute(Actor);
+
+	if (bFirstTick || OldOwnerWorkerAttribute != NetOwnerWorkerAttribute)
 	{
-		bool bOldNetOwned = bNetOwned;
+		bool bSuccess = Sender->UpdateEntityACLs(GetEntityId(), NetOwnerWorkerAttribute);
```

Comment:
It looks like this function doesn't actually provide much safety except against the ACL being nullptr because the local view saying we have authority doesn't ensure it.
Unless there a case where we try this when the worker doesn't think we have authority, as then we may not have the state corrected. But that raised the question of why we are trying to update the state of something we don't think we have authority over.

Reply:
This is only necessary for the (hopefully) rare case of having authority over `Position` but not _yet_ having received authority over `EntityACL`. We should probably have a layer that would take care of this and atomically handle perceived authority over schema components that should be together.

Reply:
That seems like something that might be intended in the not too distant future. But I agree this should be solved by some other layer and not here.